### PR TITLE
Introduced MACD chart

### LINKED ISSUE #186: MACD multi-part series plotted in another panel

### PR DESCRIPTION
**Pull request by djmiley**
Resolves #186

### CODE REVIEW

File: src/assets/js/main.js [hunk 1]
```diff
@@ -127,11 +127,10 @@
         svgPrimary.datum(dataModel)
             .call(primaryChart);
 
+        svgSecondary.selectAll('*').remove();
```
Comment (jleft) [inline src/assets/js/main.js]: Is there a way to avoid this?

File: src/assets/js/chart/macdChart.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+(function(sc) {
+    'use strict';
+    // Helper functions
```
Comment (jleft) [inline src/assets/js/chart/macdChart.js]: Remove comment

File: src/assets/js/chart/macdChart.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
+(function(sc) {
```
Comment (jleft) [inline src/assets/js/chart/macdChart.js]: Also pass in `d3` and `fc`

File: src/assets/js/chart/macdChart.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    'use strict';
+    // Helper functions
+    sc.chart.macdChart = function() {
```
Comment (jleft) [inline src/assets/js/chart/macdChart.js]: This is consistent, so we'll leave it for now, but I think I'd consider removing `Chart` from the end:

`sc.chart.macdChart ` - > `sc.chart.macd`
`sc.chart.primaryChart` -> `sc.chart.primary` 
etc.

Comment (djmiley) [inline src/assets/js/chart/macdChart.js]: Very happy to do so in this request if wanted?

Comment (jleft) [inline src/assets/js/chart/macdChart.js]: I'd leave it for now and solve it in another PR #199

File: src/assets/js/chart/macdChart.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        var dispatch = d3.dispatch('viewChange');
+
+        var macdAlgorithm = fc.indicator.algorithm.macd();
+
+        var macd = fc.indicator.renderer.macd();
+
+        function macdChart(selection) {
```
Comment (jleft) [inline src/assets/js/chart/macdChart.js]: Would we also expect this to create a Y-axis?

Comment (djmiley) [inline src/assets/js/chart/macdChart.js]: I'll work on this.

Comment (djmiley) [inline src/assets/js/chart/macdChart.js]: So the RSI component in d3fc comes 'free' with annotations but the MACD component does not.

Comment (jleft) [inline src/assets/js/chart/macdChart.js]: If the y-axis/annotations are going to be tricky to add, we could pull it out as a separate issue?

Comment (djmiley) [inline src/assets/js/chart/macdChart.js]: I think it's non-trivial so it would be a good idea.

Comment (jleft) [inline src/assets/js/chart/macdChart.js]: Okay, it can be dealt with separately - #200

**Comment by jleft** [comment]
:+1:

**Pull request merged**